src/ThemeToggle.svelte

```diff
@@ -0,0 +1,187 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+
+  let theme: "light" | "dark" | undefined = undefined;
+
+  onMount(() => {
+    let storedTheme = window.localStorage.getItem('theme');
+
+    if (storedTheme === null) {
+      // If no theme stored, set it to system default.
+      if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
+        theme = "dark";
+      } else {
+        theme = "light";
+      }
+    } else if (storedTheme === "light" || storedTheme === "dark") {
+      theme = storedTheme;
+    }
+  });
+
+  $: if (theme) {
```

Comment:
Also if we can define `theme` in the `session` store from `localStorage` or by `window.matchMedia`, then theme would never be `null`

Reply:
2b1adbfae38338e4f7bd93457e16f40ab2eb8ae5